test/integration/targets/tower_inventory_source_update/tasks/main.yml

```diff
@@ -0,0 +1,103 @@
+---
+- name: Run against a non-existant inventory
+  tower_inventory_source_update:
+    inventory: "DNE"
+    source: "DNE"
+  register: result
+  ignore_errors: True
+
+- assert:
+    that:
+      - "result is failed"
+      - "'Failed to find inventory' in result.msg"
+
+# This is directly lifted from tower_inventory_source tests
+- name: Add a Tower credential
+  tower_credential:
+    description: Credentials for Openstack Test project
+    name: openstack-test-credential
+    kind: openstack
+    organization: Default
+    project: Test
+    username: admin
+    host: https://example.org:5000
+    password: passw0rd
+    domain: test
+
+- name: Add a Tower inventory
+  tower_inventory:
+    description: Test inventory
+    name: openstack-test-inventory
+    organization: Default
+
+- name: Create a source inventory
+  tower_inventory_source:
+    name: source-test-inventory
+    description: Source for Test inventory
+    inventory: openstack-test-inventory
+    credential: openstack-test-credential
+    overwrite: True
+    update_on_launch: True
+    source_vars: "---\nprivate: false"
+    source: openstack
+  register: result
+
+- assert:
+    that:
+      - "result is successful"
+
+- name: Run againsta a non-existant inventory source
+  tower_inventory_source_update:
+    inventory: openstack-test-inventory
+    source: "DNE"
+  register: result
+  ignore_errors: True
+
+- assert:
+    that:
+      - "result is failed"
+      - "'Failed to find inventory source' in result.msg"
+
+- name: Run the inventory source update
+  tower_inventory_source_update:
+    inventory: openstack-test-inventory
+    source: source-test-inventory
+  register: result
+  ignore_errors: True
+
+# Since our credential is point to nowhere, we can only expect this to fail
+- assert:
+    that:
+      - "result is failed"
+      - "'Failed to update inventory' in result.msg"
+
+# Some day, when tower_send is imported, we can have a valid test
+#  NOTE: this has never been tested, be wary if uncommenting.
+#- name: Import our own inventory script
+#  tower_send:
+#    assets:
+#      - organization: "Default"
+#        script: "#!/bin/bash\n\necho \"{\"\necho \"}\""
+#        asset_type: "inventory_script"
+#        name: "TestScript"
+#
+#- name: Associate our inventory script with the inventory
+#  tower_inventory_source:
+#    name: source-test-script
+#    description: Source for Test inventory
+#    inventory: openstack-test-inventory
+#    overwrite: True
+#    update_on_launch: True
+#    source: custom
+#    source_path: TestScript
+#  register: result
+#
+#- name: Update an inventory source which will work
+#  tower_inventory_source_update:
+#    inventory: openstack-test-inventory
+#    source: source-test-script
+#  register: result
+#
+#- assert:
+#    that:
+#      - "result is successful"
```

Comment:
I think these comments would need to be removed.

Reply:
Done, I'm wondering if I also need to resolve the conversations.